### ISSUE TITLE
Different results when running the models manually

### ISSUE THREAD

**Issue by krb19-econ** [shankarpandala/lazypredict#366]
**Describe the bug**
I have tried using lazypredict for classification algorithms. However, when I run a certain model manually, it provides different performance metrics. 

Have attached images using the example given in the documentation.

**Screenshots**
1. Image of models from lazypredict
![image](https://user-images.githubusercontent.com/83277978/142439353-2beee189-2a31-41b3-aedd-6135f212b6ad.png)

2. Running logistic regression manually
![image](https://user-images.githubusercontent.com/83277978/142440236-a1d1326d-e0bb-4bc1-8b70-9e75f5673764.png)

**Comment by lahdjirayhan** [comment]
Have you tried supplying the same `random_state` parameters to both `LazyClassifier` and the sklearn manually-fit classifiers?

**Comment by krb19-econ** [comment]
Yes. As mentioned in the first image in the original post, the random state was 123. Using that, I have used the  same train and test datasets for both `LazyClassifier` and manual sklearn.

**Comment by lahdjirayhan** [comment]
Thanks for pointing out, but `LogisticRegression` in scikit-learn also takes a `random_state` parameter. See here: https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression

`LazyClassifier` also takes a `random_state` parameter. See here: https://github.com/shankarpandala/lazypredict/blob/652f5de3d1a21a826cac32967eb36549c8cd3b57/lazypredict/Supervised.py#L210-L218

These `random_state` parameters were not specified yet. Maybe try specifying similar values for it and see if the results are still different.

**Comment by krb19-econ** [comment]
Thanks for mentioning that. The `random_state` under `LazyClassifier` is given as 42, image attached.

![image](https://user-images.githubusercontent.com/83277978/146926981-2a08ba31-d753-4e64-9c7f-15d8cd823222.png)

I used the same under `Logistic Regression` while doing it manually. However the result is still coming out to be different. 

![image](https://user-images.githubusercontent.com/83277978/146927048-6cb04e21-6526-48b9-a5dc-e52d702b8441.png)

**Comment by shankarpandala** [comment]
It will never be same. 
lazypredict does preprocessing of data internally before doing any fit.

**Comment by nikolamilovic-rg** [comment]
I feel like this is not completely solved with the last comment. How should we treat the results of the lazy classifier then? Is it even useful if we take this into consideration? Does it use the same preprocessing steps for each of the models? Can this be at least a relative indicator for the top models (e.g. SVC gave the best accuracy after lazy classifier prediction but even though we cannot get the same accuracy by running isolated SVC model - we can say that it will give better accuracy than any other model that is executed isolated)?
